app/javascript/hooks/mutations/sessions/useCreateSession.jsx

```diff
@@ -6,13 +6,20 @@ import axios from '../../../helpers/Axios';
 export default function useCreateSession(token) {
   const queryClient = useQueryClient();
   const navigate = useNavigate();
-
+  let createRoomPermission = true;
   return useMutation(
-    (session) => axios.post('/sessions.json', { session, token }),
+    (session) => axios.post('/sessions.json', { session, token }).then((response) => {
+      createRoomPermission = response.data.data.permissions.CreateRoom;
+    }),
     {
       onSuccess: () => {
         queryClient.invalidateQueries('useSessions');
-        navigate('/rooms');
+        // if the current user does NOT have the CreateRoom permission, then do not re-direct to rooms page
+        if (createRoomPermission === 'false') {
+          navigate('/main-page');
+        } else {
+          navigate('/rooms');
+        }
```

Comment:
This is kind of weird. Can we not do all of this logic in the `onSuccess` if you just add an argument to it? `onSuccess: (data) ...`